### PR TITLE
Support NSX cert/key/ca

### PR DESCRIPTION
**Pull request by heypnus**
1. Support the user configure NSX cert/key/ca in operator
configmap yaml file
2. Node controller can connect to NSX manager via NSX cert/key

### CODE REVIEW

File: pkg/controller/node/node_controller.go [hunk 1]
```diff
@@ -187,24 +220,73 @@ func (r *ReconcileNode) createNsxClients() (*NsxClients, error) {
...
+			securityCtx.SetProperty(security.PASSWORD_KEY, password)
+			nsxClients.PolicySecurityContext = securityCtx
+			log.Info("Using username and password to connect to NSX")
+		}
+	}
+
 	// policy client
-	tlsConfig := tls.Config{InsecureSkipVerify: true}
-	tlsConfig.BuildNameToCertificate()
+	tlsConfig, err := getConnectorTLSConfig(insecure, tmpCertPath, tmpKeyPath, tmpCAPath)
```
Comment (ggverma) [inline pkg/controller/node/node_controller.go:282]: Is the `err` handled correctly here?

File: pkg/controller/node/node_controller.go [hunk 2]
```diff
@@ -187,24 +220,73 @@ func (r *ReconcileNode) createNsxClients() (*NsxClients, error) {
...
+		insecure = true
+	}
 	nsxClients := NsxClients{}
 	nsxClients.Cluster = cluster
 
+	// Get cert/key/ca, then write then into temp files
+	var certData = cfg.Section("operator").Key("nsx_api_cert").Value()
+	var keyData = cfg.Section("operator").Key("nsx_api_private_key").Value()
+	var caData = cfg.Section("operator").Key("nsx_ca").Value()
+	certDataDecoded, err := base64.StdEncoding.DecodeString(certData)
```
Comment (ggverma) [inline pkg/controller/node/node_controller.go:234]: nit: 234-245, 255-266 can make use of a func

File: manifest/cluster-network-15-nsx-secret.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 apiVersion: v1
-data: {tls.crt: null, tls.key: null}
+data: {tls.crt: "{{.NsxCert}}", tls.key: "{{.NsxKey}}", tls.ca: "{{.NsxCA}}"}
```
[the remark this answers is not in the document]
Comment (dantingl) [reply manifest/cluster-network-15-nsx-secret.yaml:2]: Yes, I think so.

File: pkg/controller/node/node_controller.go [hunk 4]
```diff
@@ -164,6 +167,36 @@ func filterSegmentPorts(nsxClients *NsxClients, ports []*data.StructValue, nodeN
...
 
+func getConnectorTLSConfig(insecure bool, clientCertFile string, clientKeyFile string, caFile string) (*tls.Config, error) {
+	tlsConfig := tls.Config{InsecureSkipVerify: insecure}
+
+	if len(clientCertFile) > 0 {
+		if len(clientKeyFile) == 0 {
+			return nil, fmt.Errorf("Please provide key file for client certificate")
+		}
+
+		cert, err := tls.LoadX509KeyPair(clientCertFile, clientKeyFile)
```
Comment (dantingl) [inline pkg/controller/node/node_controller.go:178]: I found there is another function  X509KeyPair to use PEM encoded data directly, maybe we don't need to save to a file in the middle?

File: manifest/cluster-network-15-nsx-secret.yaml (hunk 3 above)
Comment (dantingl) [inline manifest/cluster-network-15-nsx-secret.yaml:2]: Right now even operator puts CA cert in nsx-secret, but how does NCP read it? I think we lost the implementation on NCP side to read CA from secret, not sure how does QE test it before.

File: pkg/controller/node/node_controller.go (hunk 2 above)
Comment (heypnus) [reply pkg/controller/node/node_controller.go]: Done

File: pkg/controller/node/node_controller.go (hunk 1 above)
Comment (heypnus) [reply pkg/controller/node/node_controller.go:298]: Good catch, done, thanks.

File: manifest/cluster-network-15-nsx-secret.yaml (hunk 3 above)
Comment (dantingl) [reply manifest/cluster-network-15-nsx-secret.yaml:2]: Thanks, maybe later we can update this to README.

File: manifest/cluster-network-19-nsx-ncp.yaml [hunk 5]
```diff
@@ -53,3 +53,9 @@ spec:
               - {key: tls.crt, path: lb-cert/tls.crt}
               - {key: tls.key, path: lb-cert/tls.key}
               name: lb-secret
+          - secret:
```
Comment (dantingl) [inline manifest/cluster-network-19-nsx-ncp.yaml:56]: Looks like we will need to update ncp.yaml later in NCP repo.

**Comment by dantingl** [comment]
Later you should do the same like PR #24 to check if nsx-secret changes.

**Pull request merged**